app/controllers/api/admin/articles_controller.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class Api::Admin::ArticlesController < ApplicationController
+  before_action :authenticate_user!
+  before_action { check_user_role() }
+
+  def index
+    if current_user.role == 'editor'
+      article = Article.where(published: false)
+      render json: article, each_serializer: Admin::Article::IndexSerializer
+    else
+      render_unauthorized
```

Comment:
won't need this `else` if you check that the only type of user that can access this action is an editor